firestore-bigquery-export/scripts/gen-schema-view/src/index.ts

```diff
@@ -112,17 +141,71 @@ async function run(): Promise<number> {
   return 0;
 };
 
-function readSchemas(directory: string): { [schemaName: string]: FirestoreSchema } {
+function readSchemas(globs: string[]): { [schemaName: string]: FirestoreSchema } {
```

Comment:
These are fairly complicated logics. It would nice to have tests if not too hard.

Reply:
Optional nit:
Add `schemaName` to `FirestoreSchema` and return `FirestoreSchema[]` instead, which is slightly more nature to read.

You can still dedup `schemaName` internally with a separate set.

Reply:
I'd prefer to keep the `FirestoreSchema` interface as clean as possible (it should be possible to read in a json file directly into a valid FirestoreSchema object). Injecting a filesystem specific name feels like leaking information.

IMO part of the job of this script is to manage these names and I think they should probably be stored separately.